listeners/measured.go

```diff
@@ -46,11 +96,23 @@ func (c *wrapMeasuredConn) OnState(s http.ConnState) {
 // Responds to the "measured" message type
 func (c *wrapMeasuredConn) ControlMessage(msgType string, data interface{}) {
 	if msgType == "measured" {
-		c.Conn.ID = data.(string)
+		ctxUpdate := data.(map[string]interface{})
+		c.ctxMx.Lock()
+		newContext := make(map[string]interface{}, len(c.ctx))
+		// Copy context
+		for key, value := range c.ctx {
+			newContext[key] = value
+		}
+		// Update context
+		for key, value := range ctxUpdate {
+			newContext[key] = value
+		}
```

Comment:
Sorry I know this is closed @oxtoacart, but typically won't `ctxUpdate` always contain the same keys as `c.ctx` here, based on the only place this is called from being:

https://github.com/getlantern/http-proxy-lantern/pull/151/files

If so, could we just copy from `ctxUpdate` always? Or, depending on how the data is actually supposed to be updated, should we be adding to the value at that key here or something?